packages/@n8n/db/src/repositories/folder.repository.ts

```diff
@@ -373,4 +362,62 @@ export class FolderRepository extends Repository<FolderWithWorkflowAndSubFolderC
 		const result = await query.getRawMany<{ id: string }>();
 		return result.map((row) => row.id);
 	}
+
+	async getFolderPathsToRoot(folderIds: string[]): Promise<Map<string, string[]>> {
+		if (!folderIds.length) {
+			return new Map();
+		}
+
+		// Base query: select all root folders
+		const baseQuery = this.createQueryBuilder('folder')
+			.select([
+				'folder.id as id',
+				'folder.name as name',
+				'folder.parentFolderId as parentFolderId',
+				// Use CONCAT which is supported by all three databases
+				'CONCAT(folder.name) as path', // Ensures consistent type
+				'1 as level',
+			])
+			.where('folder.parentFolderId IS NULL');
+
+		// Recursive query: join child folders to their parents
+		const recursiveQuery = this.createQueryBuilder('child')
+			.select([
+				'child.id as id',
+				'child.name as name',
+				'child.parentFolderId as parentFolderId',
+				// Use CONCAT which works across all databases
+				"CONCAT(parent.path, '/', child.name) as path",
+				'parent.level + 1 as level',
+			])
+			.innerJoin('folder_paths', 'parent', 'child.parentFolderId = parent.id');
+
+		// Main query with the CTE
+		const mainQuery = this.createQueryBuilder()
+			.addCommonTableExpression(
+				`${baseQuery.getQuery()} UNION ALL ${recursiveQuery.getQuery()}`,
+				'folder_paths',
+				{ recursive: true },
+			)
+			.select('fp.id as folder_id, fp.path as folder_path')
+			.from('folder_paths', 'fp')
+			.where('fp.id IN (:...folderIds)', { folderIds });
+
+		// Execute the query
+		const results = await mainQuery.getRawMany<{
+			folder_id: string;
+			folder_path: string;
+		}>();
+
+		// Process results into the map
+		const pathMap = new Map<string, string[]>();
+
+		for (const row of results) {
+			// Split the path to get an array of folder names
+			const pathNames = row.folder_path.split('/');
+			pathMap.set(row.folder_id, pathNames);
```

Comment:
This function returns `string[]` of the path, but `FolderWithWorkflowAndSubFolderCountAndPath` expects to get just a `string` as its path, is `FolderWithWorkflowAndSubFolderCountAndPath` typed wrong and should it have `string[]` as the path field's type instead?